### PR TITLE
Update LangVersion to preview

### PR DESCRIPTION
**Pull request by hughbe**
/cc @danmosemsft

### CODE REVIEW

**Comment by zsd4yr** [comment]
Are there specific features you want?

@jaredpar it looks like [our best options for 'dogfooding'](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/langversion-compiler-option) would be to use `preview` or `latest` --do you have a preference?

**Comment by hughbe** [comment]
??= assignment mostly

File: Directory.Build.props [hunk 1]
```diff
@@ -8,6 +8,7 @@
     <Copyright>$(CopyrightNetFoundation)</Copyright>
     <PackageLicenseExpression>MIT</PackageLicenseExpression>
     <CheckForOverflowUnderflow>false</CheckForOverflowUnderflow>
+    <LangVersion>preview</LangVersion>
```
Comment (sharwell) [inline Directory.Build.props]: 📝 This should say `8.0`, not `preview`

Comment (zsd4yr) [reply Directory.Build.props]: Please make a PR to change and we can discuss what is best 😄

**Pull request merged**